compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2509,8 +2509,9 @@ public BType checkMappingConstructorCompatibility(BType bType, BLangRecordLitera
                 }
 
                 dlog.resetErrorCount();
+                BLangRecordLiteral clonedMappingConstructor = nodeCloner.cloneNode(mappingConstructor);
                 BType memCompatibiltyType = checkMappingConstructorCompatibility(listCompatibleMemType,
-                                                                                 mappingConstructor, data);
+                        clonedMappingConstructor, data);
```

Comment:
Did we consider fixing this by introducing changes similar to that for list constructors? Just to keep things consistent.

Reply:
For list constructor when silent type checking the types in the union type it seems to be setting the `data.commonAnalyzerData.nonErrorLoggingCheck = true;` [[1](https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L1740)] and using the function `exprIncompatible`[[2](https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L2242-L2245)] which clones the node in the case of `nonErrorLoggingCheck`

Reply:
Yes, what I'm asking is if we can't handle this similarly?

Reply:
Updated the code